### ISSUE TITLE
After project update to .Net Core 8  - calling MS SQL Stored Procs result in "Cannot create a DbSet"

### ISSUE THREAD

**Issue by deanparkr** [ErikEJ/EFCorePowerTools#2363]
We have recently upgraded our Net Core 6 project to Net Core 8 and have updated EF Core packages to the latest version along with the latest version of EF Power Tools.

We have noticed as part of the upgrade though that we since get this run time error when calling any/all stored procedures:

"Cannot create a DbSet for 'xxxx' because this type is not included in the model for the context.

I see this was discussed here: https://github.com/ErikEJ/EFCorePowerTools/issues/2265 but we have double checked and we are running the latest versions of everything.

Wondered what we could be missing here? We are calling the procedures like this:

```
public async Task<List<resultType>> MethodName()
{
    return await _ctx.Set<resultType>()
        .FromSqlRaw("EXECUTE xxxx")
        .ToListAsync();
}
```

- EF Core Power Tools version: 2.6.287

- Exact Visual Studio version: Visual Studio 2022 17.10.0

- Database engine: SQL Server

- EF Core version in use: Microsoft.EntityFrameworkCore.SqlServer 8.0.6

- Is Handlebars templates used: No

- Is T4 templates used: No

- Is a SQL Server .dacpac used: No

**Comment by ErikEJ** [comment]
How does this relate to EF core Power Tools?

Sounds like you have not updated the runtime, but hard to tell since you do not share any code.

**Comment by deanparkr** [comment]
It seems like issue can be resolved by adding the entities as key less entities with HasNoKey to the OnModelCreating. Eg

```
Protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        base.OnModelCreating(modelBuilder);
        
        modelBuilder.Entity<resultType>().HasNoKey();
    }
```

We can extend with a partial class to add these in for the moment. I thought as Power Tools generates the context maybe it would generate these as well. I appreciate though this change may actually be a change EF Core 8 compared to 6.

**Comment by ErikEJ** [comment]
Are you mapping stored procedures with EF Core Power Tools - it can do that for you yes  but the calling code you showet above was not from Power Tools.

**Comment by chris-ryu** [comment]
I had same symtom and after downgraded to ef 7, the problem was gone

**Comment by ErikEJ** [comment]
@chris-ryu It's a bug that was fixed in the latest 8.0.x release, I believe.
